protobuf/echo/TracingEcho.hpp

```diff
@@ -49,15 +49,15 @@ namespace services
     {}
 
     template<class T>
-    void PrintField(const T& value, services::Tracer& tracer, typename T::template Type<0>* = 0)
+    void PrintField(const T& value, services::Tracer& tracer, typename std::enable_if<!std::is_enum<T>::value>::type* = 0)
```

Comment:
The "T::template Type<0>* = 0" check checks for the presence of a Type map, which should make this function specific to messages. The new "std::enable_if<!std::is_enum<T>::value>::type* = 0" (which I expect should not be necessary, because enums don't have a Type member) is weaker; it just checks that it is not an enum, not that it is a Message. If this check is really necessary (for faulty compilers), then I'd rather have both checks, and a comment explaining why the second check is necessary.